### DISCUSSION TITLE
Break class names into multiple strings

### DISCUSSION THREAD

**Discussion by andreyvital** [tailwindlabs/tailwindcss#11514] [Ideas]
Heyo...often times I grab stuff from shadcn's coolest-ever component library, but it comes with class definitions like this:

```
"inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground"
```

You can't even see the end.

What about making this plugin split it into multiple strings concatenating?

```
"inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors" + " "
"hover:bg-muted hover:text-muted-foreground" + " "
"focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2" + " "
"disabled:pointer-events-none disabled:opacity-50" + " " 
"data-[state=on]:bg-accent data-[state=on]:text-accent-foreground" + " "
```

It would make editing so much easier.

1. I don't know the best logic to split it though, but that doesn't matter a lot at as well, if we start by breaking it by common prefixes, it'd be handy to kick-off the idea and then we figure out better grouping patterns later;

2. Ideally it should be done by this plugin so we can ensure consistency when people are editing the same code, not requiring them to have a VSCode extension installed or something;

3. Why concatenate? Because I can easily go and replace by `,` to use with `cx` utilities, but this is JS compatible for people that aren't using utility.

Thanks,
A.

**Comment by mybearworld** [comment]
Consider just using word wrap.

**Reply by mybearworld** [reply]
Although, if you have the VSCode extension and color decorators on, you'll have to disable those because they mess with word wrap for some reason

**Reply by andreyvital** [reply]
I've made an extension: https://github.com/fundamentei/tailwrap

**Reply by mybearworld** [reply]
I see this as pretty unnecessary

**Reply by andreyvital** [reply]
Well, speak for yourself—but seems very reasonable for me since I couldn't find another way to do it efficiently. I've reused some logic from `tailwind-merge` library just so it wraps the strings in a logical way. Organization is a real value for me and I do care about code "editability" ;)

**Reply by mybearworld** [reply]
It's very easy to edit code, if the person editing code just uses word wrap!

**Reply by andreyvital** [reply]
Haha, I agree. But there's a condition though: `if`. What if don't they? 🤡  That's why I created this extension—for me /not even marketing for people to use it...just sharing/, because as you can see in shadcn's component library, there's no wrapping and I don't like to waste time wrapping those. If you don't find it useful for you, that's fine. We all make different choices in life ;)

**Reply by mybearworld** [reply]
Doesn't really matter, you can just enable word wrap yourself...

**Comment by blowsie** [comment]
Duplicates
https://github.com/tailwindlabs/tailwindcss/discussions/7763

**Comment by Andrew-Pynch** [comment]
I do not understand tailwind users aversion to having classNames get wrapped automatically by prettier 🤡

**Reply by andreyvital** [reply]
Hello there, well—I am not following the topic regarding that, but I do read a lot of people into their "traditions" of not changing their mindset for some reason...the problem with that is, even though I see limitations on making certain changes that people "want" about a software, the way I see it is: if they've asked, for that person, it made sense.

For example, if you go to a pizzaria and ask for a different topping, it's because you wanted that taste...is it profitable for the pizzaria? Probably no, but then there's a surge regarding the same wish...then...it's a market opportunity.

I tend to see stuff people want as valid, just like everybody else's ideas and beliefs—just not feasible at the pace they wish sometimes.

The bottom line: **wrapping class names makes editing easier, period.** Just like having a washing machine for your clothes (so you don't have to waste a lot of time doing it yourself, bla bla...). Nobody—or at least—the majority even have mouse that features a vertical scroll so they can "go to the sides" for editing...and most people don't set up text wrapping in the editor, I find it very bad...so, how come the person above says "this is pretty unnecessary" when the reality is that he doesn't want to curve to a different mindset? (Most likely...), and it's fine...I take it, because I'm a practical and organized person, if he doesn't want to be, that's fine. Just don't block me haha.

Nuff said...like I did, if this lands in the plugin, it would make my life easier, nevertheless, I've made the extension for myself and open sourced it like all the other great minds did when they felt it would help somebody else. I'm profiting already...

**Comment by Andrew-Pynch** [comment]
Thanks Andrey, I’m excited to try your extension

From one Andrej to Another 🫡🤝

On Wed, Jul 5, 2023 at 7:46 PM Andrey Knupp Vital ***@***.***>
wrote:

> Hello there, well—I am not following the topic regarding that, but I do
> read a lot of people into their "traditions" of not changing their mindset
> for some reason...the problem with that is, even though I see limitations
> on making certain changes that people "want" about a software, the way I
> see it is: if they've asked, for that person, it made sense.
>
> For example, if you go to a pizzaria and ask for a different topping, it's
> because you wanted that taste...is it profitable for the pizzaria? Probably
> no, but then there's a surge regarding the same wish...then...it's a market
> opportunity.
>
> I tend to see stuff people want as valid, just like everybody else's ideas
> and beliefs—just not feasible at the pace they wish sometimes.
>
> The bottom line: *wrapping class names makes editing easier, period.*
> Just like having a washing machine for your clothes. Nobody even have mouse
> that features a vertical scroll so they can "go to the sides" for
> editing...and most people don't set up text wrapping in the editor, I find
> it very bad...so, how come the person above says "this is pretty
> unnecessary" when the reality is that he doesn't want to curve to a
> different mindset? (Most likely...), and it's fine...I take it, because I'm
> a practical and organized person, if he doesn't want to be, that's fine.
> Just don't block me haha.
>
> Nuff said...like I did, if this lands in the plugin, it would make my life
> easier, nevertheless, I've made the extension for myself and open sourced
> it like all the other great minds did when they felt it would help somebody
> else. I'm profiting already...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tailwindlabs/tailwindcss/discussions/11514#discussioncomment-6368581>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKKSYUWIHVIX72A6VZQNYT3XOYDIFANCNFSM6AAAAAAZZ6U6V4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
~Cheers!

**Reply by andreyvital** [reply]
Yo my friend—I didn't have time to publish it, but here's what you can do:

https://github.com/fundamentei/tailwrap/blob/master/package.json#L40


1. Clone this repository locally: `git clone https://github.com/fundamentei/tailwrap.git`;
2. Install the dependencies: `pnpm install`;
3. Run the package command: `pnpm run vsce:package`, this will generate the .vsix;
4. Run the install command: `pnpm run code:install`, this will load the raw extension into your VSCode.

Then select a class list string, including the quote, and run a command over it: "Wrap with tailwrap". That's it! ✅ 🍻 

I'm still figuring a couple of things out before the final publish, so...thats the best I could do to get going at the moment...

Thanks, you have a good week and God bless!
